apps/web/lib/features/task/task-filters.tsx

```diff
@@ -172,20 +174,22 @@ export function useTaskFilter(profile: I_UserProfilePage) {
 
 	// Set the tab to assigned if user has not planned tasks (if outstanding is empty) (on first load)
 	useEffect(() => {
-		if (dailyPlanSuggestionModalDate) {
-			if (!getTotalTasks(todayPlan)) {
-				if (estimatedTotalTime(outstandingPlans).totalTasks) {
-					setTab('dailyplan');
-				} else if (profile.tasksGrouped.assignedTasks.length) {
-					setTab('assigned');
-				} else {
-					setTab('unassigned');
+		if (dailyPlanSuggestionModalDate != new Date().toISOString().split('T')[0] && path.split('/')[1] == 'profile') {
+			if (estimatedTotalTime(outstandingPlans).totalTasks) {
+				setTab('dailyplan');
+			} else {
+				if (!getTotalTasks(todayPlan)) {
+					if (profile.tasksGrouped.assignedTasks.length) {
+						setTab('assigned');
+					} else {
+						setTab('unassigned');
+					}
 				}
 			}
 		}
 
 		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [dailyPlanSuggestionModalDate]);
+	}, []);
```

Comment:
@CREDO23 why empty array []? There are a lot of deps in the useEffect, e.g. `dailyPlanSuggestionModalDate`, todayPlan and more, do we need to add those?

Reply:
Can you resolve this Comment Please 
@CREDO23